src/Task.cc

```diff
@@ -3221,8 +3223,13 @@ static ssize_t safe_pwrite64(Task* t, const void* buf, ssize_t buf_size,
     nwritten = pwrite_all_fallible(t->vm()->mem_fd(), buf, buf_size, addr.as_int());
   }
   for (auto& m : mappings_to_fix) {
-    remote.infallible_syscall(mprotect_syscallno, m.start(), m.size(),
-                              m.prot());
+    long ret = remote.infallible_syscall_if_alive(mprotect_syscallno, m.start(),
```

Comment:
Declare this `int` so we don't need to cast it below.

Reply:
Done ! Changed the `long ret = ... ; (int)ret` pattern in another location in the same function also.